apps/ledger-live-mobile/src/screens/MyLedgerDevice/AppsScreen.tsx

```diff
@@ -238,101 +262,144 @@ const AppsScreen = ({
-          showsVerticalScrollIndicator={false}
-          contentContainerStyle={styles.list}
-        />
-      </KeyboardView>
+          ) : null}
+        </DeviceCard>
+        <ProviderWarning />
+        <Benchmarking state={state} />
+        {
+          <AppUpdateAll
+            state={state}
+            appsToUpdate={update}
+            dispatch={dispatch}
+            isModalOpened={updateModalOpened}
+          />
+        }
+      </Flex>
     ),
-
     [
-      showFwUpdateBanner,
-      onBackFromUpdate,
-      distribution,
-      state,
-      result,
+      device,
+      deviceApps,
       deviceId,
-      initialDeviceName,
-      pendingInstalls,
       deviceInfo,
       dispatch,
-      device,
-      deviceApps,
+      distribution,
+      initialDeviceName,
+      onBackFromUpdate,
+      onHeaderLayout,
       onLanguageChange,
+      pendingInstalls,
+      result,
+      showFwUpdateBanner,
+      state,
       update,
       updateModalOpened,
-      query,
-      appFilter,
-      sort,
-      order,
-      renderRow,
-      renderNoResults,
     ],
   );
 
-  const { getFeature } = useFeatureFlags();
+  const renderSearchBar = useCallback(
+    () => (
+      <Flex flexDirection="row" pb={6} backgroundColor="background.main" columnGap={space[6]}>
+        <Searchbar searchQuery={query} onQueryUpdate={setQuery} onFocus={scrollToSearchBar} />
+        <AppFilter
+          filter={appFilter}
+          setFilter={setFilter}
+          sort={sort}
+          setSort={setSort}
+          order={order}
+          setOrder={setOrder}
+        />
+      </Flex>
+    ),
+    [appFilter, order, query, scrollToSearchBar, sort, space],
+  );
+
+  const renderList = useCallback(
+    (items?: App[]) => (
+      <KeyboardView behavior="padding">
+        <SectionList
+          style={{ width: "100%", height: "100%" }}
+          ref={sectionListRef}
+          testID="manager-deviceInfo-scrollView"
+          sections={[{ data: items ?? [], title: "" }]}
+          renderItem={renderRow}
+          renderSectionHeader={renderSearchBar}
+          stickySectionHeadersEnabled
+          onScroll={onScroll}
+          ListHeaderComponent={listHeader}
+          ListFooterComponent={renderFooter(items)}
+          keyExtractor={item => item.name}
+          showsVerticalScrollIndicator={false}
+          contentContainerStyle={styles.list}
+        />
+      </KeyboardView>
+    ),
+    [listHeader, onScroll, renderFooter, renderRow, renderSearchBar],
+  );
+
+  const { getFeature, isFeature } = useFeatureFlags();
   const enabledApps = useMemo(
     () =>
       catalog.filter(({ currencyId }: App) => {
         if (!currencyId) return true;
         const currencyFeatureKey = camelCase(`currency_${currencyId}`) as FeatureId;
-        return getFeature(currencyFeatureKey)?.enabled ?? true;
+        return isFeature(currencyFeatureKey) ? getFeature(currencyFeatureKey)?.enabled : true;
```

Comment:
Note: this is unrelated to the main point of this ticket but I was just tired of the 100+ error messages whenever opening My Ledger